### ISSUE TITLE
Crash with Google Play Services 8.3.0

### ISSUE THREAD

**Issue by gyulavoros** [couchbase/couchbase-lite-android#695]
Upgrading to GPS 8.3.0 requires you to bump the gradle build tools as well to `com.android.tools.build:gradle:1.5.0-beta2` (or your project won't even compile). After upgrading, Couchbase crashes with the following exception:

```
AndroidRuntime: FATAL EXCEPTION: main
    Process: hu.gyulavoros.cbcrashsample, PID: 7365
        java.lang.UnsatisfiedLinkError: dalvik.system.PathClassLoader[DexPathList[[zip file "/data/app/hu.gyulavoros.cbcrashsample-2/base.apk"],nativeLibraryDirectories=[/vendor/lib, /system/lib]]] couldn't find "libcom_couchbase_touchdb_TDCollateJSON.so"
        at java.lang.Runtime.loadLibrary(Runtime.java:366)
        at java.lang.System.loadLibrary(System.java:989)
        at com.couchbase.touchdb.TDCollateJSON.<clinit>(TDCollateJSON.java:36)
        at com.couchbase.lite.android.AndroidSQLiteStorageEngine.open(AndroidSQLiteStorageEngine.java:57)
        at com.couchbase.lite.Database.open(Database.java:977)
        at com.couchbase.lite.Manager.getDatabase(Manager.java:243)
        at hu.gyulavoros.cbcrashsample.BaseApplication.onCreate(BaseApplication.java:14)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1020)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5109)
        at android.app.ActivityThread.access$1600(ActivityThread.java:177)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1509)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:145)
        at android.app.ActivityThread.main(ActivityThread.java:5938)
        at java.lang.reflect.Method.invoke(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:372)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1400)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1195)
```

I created an empty sample project demonstrating the issue: https://github.com/gyulavoros/cb-crash-sample

You can change build tools versions by switching the comments in https://github.com/gyulavoros/cb-crash-sample/blob/master/build.gradle

I have limited experience with the NDK, but after Googling around I think it's possible that the issue is related to the build tools and not to couchbase-lite. (It looks like the guys at Realm struggle with similar issues).

Any help would be appreciated!

**Comment by cujo** [comment]
AFAIK, this issue is not related to CBL and will be fixed in 1.5.0-beta3

**Comment by gyulavoros** [comment]
1.5.0-beta3 released and the issue is gone.
